persist/fs/commitlog/writer.go

```diff
@@ -191,9 +213,9 @@ func (w *writer) Write(
 		return err
 	}
 
-	if !seen {
-		// Record we have seen this series
-		w.bitset.set(logEntry.Index)
+	if !definitelySeen {
+		// Record we have seen this series at the current seenIdx
+		w.seenCache[seenIdx] = logEntryIndex
```

Comment:
We only want to add it to the `seenCache` if there was no error writing it to the log.